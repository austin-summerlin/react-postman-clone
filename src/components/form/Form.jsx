/* eslint-disable max-len */
import React from 'react';
import PropTypes from 'prop-types';
import RadioButtons from '../radio/Radio';
import style from '../form/style.css';

const Form = ({ url, body, method, onSubmit, onChange }) => {
  return (
    <section className={style.form}>
      <form onSubmit={onSubmit}>
        <input type="text" name="url" placeholder="URL" value={url} onChange={onChange} />
        <section>
          <RadioButtons method={method} onChange={onChange} />
          <button>Go</button>
        </section>
        <textarea name="body" placeholder="Raw JSON Body" value={body} onChange={onChange} />
      </form>
    </section>
  );
};

Form.propTypes = {
  onSubmit: PropTypes.func.isRequired,
  method: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  url: PropTypes.string.isRequired,
  body: PropTypes.any.isRequired,
};

export default Form;






