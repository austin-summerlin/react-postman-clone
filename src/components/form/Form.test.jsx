import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Form from '../form/Form';

describe('Form Component', () => {
  afterEach(() => cleanup());
  it.only('should render Form', () => {
    const { asFragment } = render(<Form />);
    expect(asFragment()).toMatchSnapshot();
  });
});
