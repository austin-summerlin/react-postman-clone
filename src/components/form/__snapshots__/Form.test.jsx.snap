// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form Component should render Form 1`] = `
<DocumentFragment>
  <section
    class="form"
  >
    <form>
      <input
        name="url"
        placeholder="URL"
        type="text"
        value=""
      />
      <section>
        <div>
          <input
            id="get"
            name="method"
            type="radio"
            value="GET"
          />
          <label
            for="get"
          >
            GET
          </label>
          <input
            id="post"
            name="method"
            type="radio"
            value="POST"
          />
          <label
            for="post"
          >
            POST
          </label>
          <input
            id="put"
            name="method"
            type="radio"
            value="PUT"
          />
          <label
            for="put"
          >
            PUT
          </label>
          <input
            id="delete"
            name="method"
            type="radio"
            value="DELETE"
          />
          <label
            for="delete"
          >
            DELETE
          </label>
        </div>
        <button>
          Go
        </button>
      </section>
      <textarea
        name="body"
        placeholder="Raw JSON Body"
      />
    </form>
  </section>
</DocumentFragment>
`;
